DataServer/runServer.py

```diff
@@ -41,17 +42,20 @@ def main():
     root_logger.addHandler(console_handler)
     logging.Formatter.converter = time.gmtime
 
+    def report_outage(device: Device):
+        root_logger.warn(f'There was an outage: device {device.deviceID} went offline')
```

Comment:
Maybe this should live inside the `Server` class?  This seems to be inherent to the Server behavior.